proposals/extensions.md

```diff
@@ -50,6 +50,48 @@ receiver_parameter // add
 Extension declarations shall only be declared in non-generic, non-nested static classes.  
 It is an error for a type to be named `extension`.  
 
+### Declaration spaces
+
+Section [7.3 Declarations](https://github.com/dotnet/csharpstandard/blob/standard-v7/standard/basic-concepts.md#73-declarations) in the C# Standard is updated as follows (additions in **bold non-italic**):
+
+> ...
+> 
+> There are several different types of declaration spaces, as described in the following.
+> 
+> ...
+> 
+> - Each method declaration, property declaration, property accessor declaration, indexer declaration, indexer accessor declaration, operator declaration, instance constructor declaration, anonymous function, and local function creates a new declaration space called a ***local variable declaration space***. Names are introduced into this declaration space through formal parameters (*fixed_parameter*s and *parameter_array*s) and *type_parameter*s. The set accessor for a property or an indexer introduces the name `value` as a formal parameter. 
+**An *extension_declaration* introduces to each member declaration directly contained within it its *type_parameter_list* as type parameters and the *identifier*, if any, of its *receiver_parameter* as a formal parameter.**
```

Comment:
I think this rule can lead to unfriendly typing experience. Let's say I want to declare a method returning ```T``` in ```extension<T>(T[] x)```. Completion list in IDE won't offer ```T```. And if I type ```T``` and then press space or tab, completion is likely to insert something else that starts with "T". That is because ```T``` is not in scope, we are not in context of any member declaration yet.

Reply:
Also, are type parameters of extension in scope inside receiver parameter declaration? Are they in scope inside constraints?

Reply:
I don't think we have to implement it that way! I think the closest analogy is query expressions. From a spec point of view, the variable introduced by e.g. a `from` clause doesn't "exist" but becomes a parameter of several subsequent lambda expressions in a syntactic rewrite. However, the tooling experience treats it as a single variable. I'm imagining we do the same kind of thing here.

Reply:
> I don't think we have to implement it that way!

I do not think this answers the following questions asked earlier on this thread: "Also, are type parameters of extension in scope inside receiver parameter declaration? Are they in scope inside constraints?"

> I think the closest analogy is query expressions.

I am sorry, but I do not get the analogy. 

> I'm imagining we do the same kind of thing here.

I am not sure what is being imagined here.

I suggest we get together for a half an hour an discuss this offline in more details.

Reply:
I've updated the rules based on feedback. They are no longer expressed as a delta on the language spec, but more directly describe the intended behavior.